core/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -192,14 +193,26 @@ public Translog(
                 final long generation = deletionPolicy.getMinTranslogGenerationForRecovery();
                 logger.debug("wipe translog location - creating new translog, starting generation [{}]", generation);
                 Files.createDirectories(location);
-                final Checkpoint checkpoint = Checkpoint.emptyTranslogCheckpoint(0, generation, globalCheckpointSupplier.getAsLong());
+                final Checkpoint checkpoint = Checkpoint.emptyTranslogCheckpoint(0, generation, globalCheckpointSupplier.getAsLong(), generation);
                 final Path checkpointFile = location.resolve(CHECKPOINT_FILE_NAME);
                 Checkpoint.write(getChannelFactory(), checkpointFile, checkpoint, StandardOpenOption.WRITE, StandardOpenOption.CREATE_NEW);
                 IOUtils.fsync(checkpointFile, false);
-                current = createWriter(generation);
-
+                current = createWriter(generation, new LongSupplier() {
+                    // this is a bit of a nasty hack to work around the fact that we're not fully initialized
```

Comment:
This is so disgusting, can we simply push down two parameters, one is the initial value (as a long) and `translog::getMinFileGeneration` as the supplier?